rsconnect/api.py

```diff
@@ -360,7 +360,7 @@ def __init__(
         url: str = None,
         api_key: str = None,
         insecure: bool = False,
-        cacert: IO = None,
```

Comment:
cacert should be handled as a file, hence IO; whereas cadata is text, hence str

Reply:
Correct. Here the cacert variable is a string representing the file path, instead of a file pointer. The call on line 436 opens the file pointer, which was originally handled by Click), and reads the file contents in either text mode or byte mode depending on the file-type.

The existing implementation assumes that the file pointer has been opened in text mode, which doesn't work when using DER files since they are encoded in binary format. L436 onwards, cadata is either a `str` or `bytes`, both of which are accepted when creating an SSLContext. This is the same as the existing implementation.

The Click documentation goes into scenario a little further: https://click.palletsprojects.com/en/8.1.x/arguments/#file-path-arguments